languages/csharp-dotnet-2.2/template/src/pipe.cs

```diff
@@ -0,0 +1,16 @@
+using Algorithmia;
+using __ALGO__;
+// This file takes your __ALGO__.cs file, loads the apply function and creates a Pipe wrapper around it.
+// WARNING: Do not remove this file from your project, it will make your algorithm unusable.
+
+namespace __ALGO__
+
+{
+    public class pipe
+    {
+        static int Main()
+        {
+            return AlgorithmiaPipe.Pipe.Enter(typeof(__ALGO__));
```

Comment:
I was just looking at the magic for `AlgorithmiaPipe`, and while the comment probably belongs in that repo, I don't grok how an algorithm would actually accept binary data. Would the algo itself be responsible for decoding the base64 string?

Reply:
nope, the pipe deserializes the string into a byte array, you can see it here:
https://github.com/algorithmiaio/algorithmia-c-sharp-dev/blob/master/src/Module.cs#L75

Reply:
Right now the user is responsible of serializing binary data back into base 64 encoded strings, we could add automatic support for this later - but we don't support that now, so /shrug.